crates/bevy_render/src/view/window.rs

```diff
@@ -121,6 +121,16 @@ pub struct WindowSurfaces {
     configured_windows: HashSet<WindowId>,
 }
 
+/// Creates or updates window surfaces for rendering.
+///
+/// NOTE: `prepare_windows` can take a long time (and this can be seen using profiling)
+/// if the GPU is a serious bottleneck and doesn't manage to keep up with CPU.
+/// This can be caused by many reasons, but several of them are:
+/// - Just too much load for current spec
+/// - Unusual high workload (possibly an error in your custom shaders)
+/// - Bad drivers or mismatch with chosen [`Backends`](crate::settings::Backends) /
+///   [`WgpuLimits`](crate::settings::WgpuLimits) / [`WgpuFeatures`](crate::settings::WgpuFeatures)
+///   causing `wgpu` to choose a (very slow) software renderer adapter.
```

Comment:
Here's my attempt. It is verbose but I tried to include the useful information that I have understood from your testing and our discussions.
```suggestion
/// NOTE: `get_current_texture` in `prepare_windows` can take a long time if the GPU workload is
/// the performance bottleneck. This can be seen in profiles as multiple prepare-stage systems all
/// taking an unusually long time to complete, and all finishing at about the same time as the
/// `prepare_windows` system. Improvements in bevy are planned to avoid this happening when it
/// should not but it will still happen as it is easy for a user to create a large GPU workload
/// relative to the GPU performance and/or CPU workload.
/// This can be caused by many reasons, but several of them are:
/// - GPU workload is more than your current GPU can manage
/// - Error / performance bug in your custom shaders
/// - wgpu was unable to detect a proper GPU hardware-accelerated device given the chosen
///   [`Backends`](crate::settings::Backends), [`WgpuLimits`](crate::settings::WgpuLimits),
///   and/or [`WgpuFeatures`](crate::settings::WgpuFeatures). For example, on Windows currently
///   DirectX 11 is not supported by wgpu 0.12 and so if your GPU/drivers do not support Vulkan,
///   it may be that a software renderer called "Microsoft Basic Render Driver" using DirectX 12
///   will be chosen and performance will be very poor. This is visible in a log message that is
///   output during renderer initialization. Future versions of wgpu will support DirectX 11, but
///   another alternative is to try to use [`ANGLE`](https://github.com/gfx-rs/wgpu#angle) and
///   [`Backends::GL`](Backends::GL) if your GPU/drivers support OpenGL 4.3 / OpenGL ES 3.0 or
///   later.
```
Though maybe the information at the end would be better documented (also?) elsewhere. @alice-i-cecile do you have some good ideas where advice about how to configure bevy to perform well on some platform or other should go? It feels like a getting started kind of thing and I imagine we have something about the wasm + WebGL2 target for example requiring the `webgl` feature. This is on a similar level to that, if a bit more specific.

Reply:
This ultimately belongs in a Performance Optimization chapter of the book.

Reply:
I think the part specifically about ‘if your gpu/drivers don’t support dx12/vulkan(/metal) then these are your options’ is more of an initial concern perhaps. The other part about GPU workload, I agree is deeper performance stuff.

Reply:
I think that this level of verbosity will be very helpful to anyone encountering similar problems 🧇 . I also thought of something like writing this kind of stuff (and about choosing backends above) to some kind of guide, in md or bevy book, this can be very useful I think ;)